tests/time_tests/run_executable.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
+# Copyright (C) 2020 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""
+This script runs TimeTests executable several times to aggregate
+collected statistics.
+"""
+
+# pylint: disable=redefined-outer-name
+
+import statistics
+from pathlib import Path
+import tempfile
+import subprocess
+import logging
+import argparse
+import sys
+from pprint import pprint
+import yaml
+
+
+def run_cmd(args: list, log=None, verbose=True):
+    """ Run command
+    """
+    if log is None:
+        log = logging.getLogger('run_cmd')
+    log_out = log.info if verbose else log.debug
+
+    log.info(f'========== cmd: {" ".join(args)}')  # pylint: disable=logging-fstring-interpolation
+
+    proc = subprocess.Popen(args,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.STDOUT,
+                            encoding='utf-8',
+                            universal_newlines=True)
+    output = []
+    for line in iter(proc.stdout.readline, ''):
+        log_out(line.strip('\n'))
+        output.append(line)
+        if line or proc.poll() is None:
+            continue
+        break
+    outs = proc.communicate()[0]
+
+    if outs:
+        log_out(outs.strip('\n'))
+        output.append(outs)
+    log.info('========== Completed. Exit code: %d', proc.returncode)
+    return proc.returncode, ''.join(output)
+
+
+def read_stats(stats_path, stats: dict):
+    """Read statistics from a file and extend provided statistics"""
+    with open(stats_path, "r") as file:
+        parsed_data = yaml.load(file, Loader=yaml.FullLoader)
+    return dict((step_name, stats.get(step_name, []) + [duration])
+                for step_name, duration in parsed_data.items())
+
+
+def aggregate_stats(stats: dict):
+    """Aggregate provided statistics"""
+    return {step_name: {"avg": statistics.mean(duration_list),
+                        "stdev": statistics.stdev(duration_list)}
+            for step_name, duration_list in stats.items()}
+
+
+def write_aggregated_stats(stats_path, stats: dict):
+    """Write aggregated statistics to a file in YAML format"""
+    with open(stats_path, "w") as file:
+        yaml.dump(stats, file)
+
+
+def prepare_executable_cmd(args: dict):
+    """Generate common part of cmd from arguments to execute"""
+    return [str(args["executable"].resolve()),
+            "-m", str(args["model"].resolve()),
+            "-d", args["device"]]
+
+
+def run_executable(args: dict, log=None):
+    """Run provided executable several times and aggregate collected statistics"""
+
+    if log is None:
+        log = logging.getLogger('run_executable')
+
+    cmd_common = prepare_executable_cmd(args)
+
+    # Run executable and collect statistics
+    stats = {}
+    for run_iter in range(args["niter"]):
+        tmp_stats_file = tempfile.NamedTemporaryFile()
```

Comment:
This may not work on Windows - application under run_cmd may fail to open the file the second time (note it is already being opened by NamedTemporaryFile) https://stackoverflow.com/questions/17502704/python-tempfile-how-to-open-several-times-the-temp-file

Reply:
Good catch, it really causes `Permission Denied` on Win. I resolved it with manual deletion of temp file before passing it to executable